src/core/lib/transition-manager.js

```diff
@@ -155,6 +161,20 @@ export default class TransitionManager {
     this.state = DEFAULT_STATE;
   }
 
+  // Update angle props so they take shortest interpolation path.
+  _updateAngles(startViewport, endViewport) {
```

Comment:
This should be moved into the `viewportLinearInterpolator`.
This is the only code in `TransitionManager` that is `WebMercatorViewport` specific.
Also since this is really only needed by `viewportLinearInterpolator`, hard coding it here may become a trap for custom interpolators.

Reply:
I don't think this is specific to any interpolator. Any interpolation of angles needs this, may be now specific to `WebMercatorViewport`, but in future if we add transitions for`OribitViewport` or any other viewport that contains angles, we need it.